pootle/apps/pootle_store/models.py

```diff
@@ -865,17 +865,21 @@ def toggle_qualitycheck(self, check_id, false_positive, user):
             sub_type = SubmissionTypes.MUTE_CHECK
         else:
             sub_type = SubmissionTypes.UNMUTE_CHECK
-
-        sub = Submission(creation_time=make_aware(timezone.now()),
-                         translation_project=self.store.translation_project,
-                         submitter=user, field=SubmissionFields.NONE,
-                         unit=self, store=self.store, type=sub_type,
-                         quality_check=check)
+        update_time = make_aware(timezone.now())
+        sub = Submission(
+            creation_time=update_time,
+            translation_project=self.store.translation_project,
+            submitter=user,
+            field=SubmissionFields.NONE,
+            unit=self,
+            store=self.store,
+            type=sub_type,
+            quality_check=check)
         sub.save()
 
         # update timestamp
         # log user action
-        self.save()
+        self.save(submitted_by=user, submitted_on=update_time)
```

Comment:
@ta2-1 atm - there is a lot of assumptions about what does/not happen - the tests that have been added this week have been to challenge those assumptions and in code at least document how these things worked.

I would be really up for some docs around this - but for now my concern is ensuring it does what we expect, and our expectations have been thought through